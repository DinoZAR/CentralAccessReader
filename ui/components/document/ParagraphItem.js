import React from 'react';
import PropTypes from 'prop-types';

const TextItem = ({ content }) => {
  return <span>{content.content}</span>;
};

const ImageItem = ({ content }) => {
  return <img src={content.url} />;
};

const typesToItems = {
  text: TextItem,
  image: ImageItem,
};
const ParagraphItem = ({ content }) => {
  const Comp = typesToItems[content.type];
  return <Comp content={content} />;
};

export default ParagraphItem;
