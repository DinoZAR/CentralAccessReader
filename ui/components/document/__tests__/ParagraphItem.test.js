import React from 'react';
import ParagraphItem from '../ParagraphItem';

describe('ParagraphItem', () => {
  it('renders text', () => {
    const wrapper = shallow(
      <ParagraphItem
        content={{
          type: 'text',
          content: 'Here is some text.'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders images', () => {
    const wrapper = shallow(
      <ParagraphItem
        content={{
          type: 'image',
          url: 'some_image.png'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
