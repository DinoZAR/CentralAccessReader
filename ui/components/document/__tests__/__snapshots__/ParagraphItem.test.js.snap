// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParagraphItem renders images 1`] = `
<ImageItem
  content={
    Object {
      "type": "image",
      "url": "some_image.png",
    }
  }
/>
`;

exports[`ParagraphItem renders text 1`] = `
<TextItem
  content={
    Object {
      "content": "Here is some text.",
      "type": "text",
    }
  }
/>
`;
